packages/beacon-node/src/network/options.ts

```diff
@@ -11,6 +11,12 @@ export interface INetworkOptions extends PeerManagerOpts, ReqRespBeaconNodeOpts,
   mdns: boolean;
   connectToDiscv5Bootnodes?: boolean;
   version?: string;
+
+  // TODO: These options  are deprecated and will be removed in next major release.
+  requestCountPeerLimit?: number;
+  blockCountPeerLimit?: number;
+  blockCountTotalLimit?: number;
+  rateTrackerTimeoutMs?: number;
```

Comment:
Let me know your opinion, but currently we do API breakages for programmatic usage options. I would argue to keep breaking until https://github.com/ChainSafe/lodestar/pull/4576 so I'm in favor of dropping the options from here, but keep the CLI flags marked as deprecated

Reply:
Yes I agree, we can remove the options and just keep the flags. As the CLI value parsing actually don't have any impact. To not break the CLI scripts user may have, we can keep the CLI options.

Reply:
For reference it was kept because of the code usage. 

```ts
defaultDescription: String(defaultOptions.network.rateTrackerTimeoutMs),
```

Reply:
Yes, for deprecated options I would just keep the minimum code necessary to not make yargs throw an error